doc/algorithms.md

```diff
@@ -0,0 +1,727 @@
+# Core Algorithms used in MeetEval
+
+This file contains simplified implementations of the core algorithms used in the MeetEval toolkit.
+These implementations are meant for documentation.
+The actual implementations in the toolkit are more complex and track the alignment along with the distance computation, which is omitted here for clarity.
+
+All algorithms in this file operate on a _character level_ instead of the word level.
+This doesn't change the algorithms by themselves, but simplifies the implementations (because `str.split()` can be omitted).
+Keep in mind that these simplified implementations may not cover all edge-cases correctly.
```

Comment:
Do you know if one algorithm has issues in an edge case? If yes, we should add a folding comment to that algorithm.

We should remove this statement.